source/drivers/linux/chipsec_km.c

```diff
@@ -1568,29 +1486,32 @@ int find_symbols(void)
 int __init
 init_module (void)
 {
-	int sym_status = 0;
+	int ret = 0;
 	printk(KERN_ALERT "Chipsec module loaded \n");
 	printk(KERN_ALERT "** This module exposes hardware & memory access, **\n");
 	printk(KERN_ALERT "** which can effect the secure operation of      **\n");
 	printk(KERN_ALERT "** production systems!! Use for research only!   **\n");
 
-	sym_status = find_symbols();
-	chr_dev_init();  
-	if(sym_status) 
+	ret = find_symbols();
+	if (ret)
 	{
 		printk("Chipsec symbol lookup failed\n");
-		cleanup_module();
 		return -1;
 	}
+	ret = misc_register(&chipsec_dev);
+	if (ret)
+	{
+		printk("Unable to register the chipsec divice\n");
```

Comment:
Thanks, fixed in https://github.com/chipsec/chipsec/pull/29/commits
If we return a value that is non-zero, the userland caller (insmod) will print an error and translate it to plain English: http://git.kernel.org/cgit/utils/kernel/module-init-tools/module-init-tools.git/tree/insmod.c#n161

I'd rather rely on that for informing the user than print an integer or interpret it from the kernel.